aip/general/0162.md

```diff
@@ -25,201 +25,170 @@ to the version of an API as a whole.
 
-**Note:** A randomly generated string is preferred over other methods, such as
-an auto-incrementing integer, because there is often a need to delete or revert
-revisions, and a randomly generated string holds up better in those situations.
+### Creating Revisions
 
-### Referencing revisions
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-When it is necessary to refer to a specific revision of a resource, APIs
-**must** use the following syntax: `{resource_name}@{revision_id}`. For
-example:
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
+- Create a new revision when the user specifically asks
 
-    publishers/123/books/les-miserables@c7cfa2a8
+APIs **may** use any of these strategies.
 
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
+### Resource names for revisions
 
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+When referring to specific revision of a resource, the subcollection name
+**must** named `revisions`. Resource revisions have names with the format
+`{resource_name}/revisions/{revision_id}`. For example:
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+publishers/123/books/les-miserables/revisions/c7cfa2a8
 
-### Getting a revision
+### Server specified IDs
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+Services **may** reserve specific IDs to be server-specified (e.g. `current`,
+`latest`), which are read-only and managed by the service.
 
-```proto
-message GetBookRequest {
-  // The name of the book.
-  //   Example: publishers/123/books/les-miserables
-  //
-  // In order to retrieve a previous revision of the book, also provide
-  // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
-  string name = 1 [
-    (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
-    }];
-}
+```
+GET /v1/publishers/{publisher}/books/{book}/revisions/{revision_id}
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+- If a `latest` ID exists, it **must** represent the most recently created
```

Comment:
I've just removed the mention of a `current` ID, since `latest` is the only well defined one. The old version didn't have a concept of `current` ID anyway.

Reply:
Cool. I'll have another look through all the changes tomorrow.